test/inference/smc.jl

```diff
@@ -0,0 +1,17 @@
+using Turing, Random, Test
+using StatsFuns
+
+dir = splitdir(splitdir(pathof(Turing))[1])[1]
+include(dir*"/test/test_utils/AllUtils.jl")
+
+@turing_testset "smc.jl" begin
+  @model normal() = begin
+    a ~ Normal(4,5)
+    3 ~ Normal(a,2)
+    b ~ Normal(a,1)
+    1.5 ~ Normal(b,2)
+    a, b
+  end
+
+  tested = sample(normal(), SMC(10))
```

Comment:
Maybe add a failing test example?

Reply:
The above example (in PR description) will fail, but I'm not sure how to test it. Any suggestions?

Reply:
Would it work with the `@test_throws` macro?